llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp

```diff
@@ -1159,7 +1159,9 @@ class ARMOperand : public MCParsedAsmOperand {
     if (!isImm()) return false;
     const MCConstantExpr *CE = dyn_cast<MCConstantExpr>(getImm());
     if (!CE) return false;
```

Comment:
Can this check that CE->getActiveBits() > 32?

Reply:
Not quite, but this put me on the right path. I added a check for `isUInt<32>` and dropped the signed flag.